### DISCUSSION TITLE
Lesson 26: Testing events

### DISCUSSION THREAD

**Discussion by r0bchain** [Cyfrin/foundry-full-course-cu#1969] [General]
Hi there, I am trying to understand this piece of code:

```
 function testEnteringRaffleEmitsEvent() public {
        //  Arrange
        vm.prank(PLAYER);

        //  Act
        vm.expectEmit(true, false, false, false, address(raffle));
        emit RaffleEnter(PLAYER);

        //  Assert 
        raffle.enterRaffle{value: entranceFee}();
    }
```
But we are also calling this emit event here:
```
function enterRaffle() public payable {
        if (msg.value < i_entranceFee) {
            revert Raffle__SendMoreToEnterRaffle();
        }
        if (s_raffleState != RaffleState.OPEN) {
            revert Raffle__RaffleNotOpen();
        }
        s_players.push(payable(msg.sender));
        emit RaffleEnter(msg.sender);
    }
```
As I can see, the method enterRaffle is also calling the emit event RaffleEnter. Does this mean we are emitting the event twice? I'm a bit confused.

**Comment by EngrPips** [comment]
Hello @r0bertinski, You are correct. The contract would emit an event when you call the `enterRaffle` function, but there is no way for us to catch the event inside our test contract, so we define an event in our test and emit it. So, we then capture the event we emitted in our test to check and confirm that the event is properly emitted.
